server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -57,6 +57,8 @@
 
     private static final Logger logger = LogManager.getLogger(RestController.class);
     private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(RestController.class);
+    private static final String ELASTIC_PRODUCT_HTTP_HEADER = "X-Elastic-Product";
```

Comment:
Should this be lowercase here? I noticed that many of the HTTP Headers we operate with are not consistently cased.

Reply:
According to the HTTP spec, header names are case-insensitive. We use lowercase for `x-elastic-product-origin` but it shouldn't* matter.

`*` famous last words